frontend/tests/badges/BadgesTest.php

```diff
@@ -247,4 +245,28 @@ public function testAssignBadgesCronjob() {
             );
         }
     }
+
+    public function testGetAssignationTime() {
+        $user = UserFactory::createUser();
+        ProblemsFactory::createProblemWithAuthor($user);
+
+        $previousTime = Time::get();
+        Utils::RunAssignBadges();
+
+        $login = self::login($user);
+        $problemSetterResult = BadgeController::apiMyBadgeAssignationTime(new Request([
+            'auth_token' => $login->auth_token,
+            'user' => $user,
+            'badge_alias' => 'problemSetter',
+        ]));
+        $timeDifference = ($problemSetterResult['assignation_time'] - $previousTime) / 60;
+        $this->assertTrue($timeDifference < 10);
+
+        $contestManagerResult = BadgeController::apiMyBadgeAssignationTime(new Request([
+            'auth_token' => $login->auth_token,
+            'user' => $user,
+            'badge_alias' => 'contestManager',
+        ]));
+        $this->assertFalse($contestManagerResult['assignation_time']);
```

Comment:
nit: esto no es un booleano (conceptualmente), así que hay que usar [`assertNull`](https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.assertNull) y corregir el API para que regrese un entero o nulo.

```suggestion
        $this->assertNull($contestManagerResult['assignation_time']);
```